test/EmbeddedTests/SmugglerTests.cs

```diff
@@ -0,0 +1,197 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Threading.Tasks;
+using Raven.Client.Documents.Smuggler;
+using Raven.Client.Exceptions;
+using Raven.Embedded;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+using Xunit;
+// ReSharper disable UseAwaitUsing
+// ReSharper disable IdentifierTypo
+
+namespace EmbeddedTests
+{
+    public class SmugglerTests : EmbeddedTestBase
+    {
+        [Fact]
+        public async Task SmugglerImportFileShouldThrowTimeout()
+        {
+            const string fileName = "dump.cmp";
+
+            var paths = CopyServer();
+
+            using (var embedded = new EmbeddedServer())
+            {
+                embedded.StartServer(new ServerOptions
+                {
+                    ServerDirectory = paths.ServerDirectory,
+                    DataDirectory = paths.DataDirectory,
+                });
+
+                const string databaseName = "test";
+                var dummyDump = CreateDummyDump(100000);
+
+                using (var ctx = JsonOperationContext.ShortTermSingleUse())
+                using (var bjro = ctx.ReadObject(dummyDump, "dump"))
+                using (var ms = File.Create(fileName))
+                using (var s = new GZipStream(ms, CompressionMode.Compress))
+                {
+                    bjro.WriteJsonTo(s);
+                }
+
+                using (var store = embedded.GetDocumentStore(new DatabaseOptions(databaseName)))
+                {
+                    using (store.SetRequestTimeout(TimeSpan.FromMilliseconds(1000), databaseName))
+                    {
+                        try
+                        {
+                            var operation = await store.Smuggler.ForDatabase(databaseName).ImportAsync(new DatabaseSmugglerImportOptions
+                            {
+                                OperateOnTypes = DatabaseItemType.Documents | DatabaseItemType.Identities | DatabaseItemType.CompareExchange,
+                            }, "dump.cmp");
+
+                            await operation.WaitForCompletionAsync();
```

Comment:
You are not failing the test here. After this line you should add an assert or throw that it should not happen